client/src/components/Modal.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef } from "react";
+
+type ModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  children: React.ReactNode;
+  className?: string;
+};
+
+export default function Modal({
+  onClose,
+  isOpen,
+  children,
+  className,
+}: ModalProps) {
+  const modalRef = useRef<HTMLDialogElement | null>(null);
+
+  useEffect(() => {
+    const { current: el } = modalRef;
+
+    if (!el) return;
+
+    if (isOpen) {
+      el.showModal();
+    } else {
+      el.close();
+    }
+  }, [isOpen]);
+  return (
+    <dialog ref={modalRef} className={className}>
+      <button
+        onClick={onClose}
+        style={{
```

Comment:
I tend to keep my inline styles to less than 3 - what do you think of using that as our standard for the project, and moving these styles into a css file?

Reply:
moving these still out of here and passing the classes in as a prop instead.

Reply:
Awesome, thanks!